temporian/implementation/numpy/data/display_utils.py

```diff
@@ -0,0 +1,199 @@
+                data_repr.append(
+                    f"... (showing {max_indexes} of {len(evset.data)} indexes)"
+                )
+                break
+            index_key_repr = []
+            for index_value, index_name in zip(
+                index_key, evset.schema.index_names()
+            ):
+                index_key_repr.append(f"{index_name}={index_value}")
+            index_key_repr = " ".join(index_key_repr)
+            data_repr.append(
+                f"{index_key_repr} ({len(index_data.timestamps)} events):\n"
+                f"    timestamps: {index_data.timestamps}\n"
+                f"{string.indent(repr_features(evset, index_data.features))}"
+            )
+        data_repr = string.indent("\n".join(data_repr))
+
+    return (
+        f"indexes: {evset.schema.indexes}\n"
+        f"features: {evset.schema.features}\n"
+        "events:\n"
+        f"{data_repr}\n"
+        f"memory usage: {string.pretty_num_bytes(evset.memory_usage())}\n"
+    )
+
+
+def display_html(evset: EventSet) -> str:
+    """HTML representation, mainly for IPython notebooks."""
+    from xml.dom import minidom
+
+    def create_table_row(
+        columns: List[str], header: bool = False
+    ) -> minidom.Element:
+        tr = dom.createElement("tr")
+        for col in columns:
+            if header:
+                td = dom.createElement("th")
+                text = dom.createElement("b")
+                text.appendChild(dom.createTextNode(col))
+            else:
+                td = dom.createElement("td")
+                text = dom.createTextNode(col)
+            td.appendChild(text)
+            tr.appendChild(td)
+        return tr
+
+    # Create DOM
+    impl = minidom.getDOMImplementation()
+    assert impl is not None
+    dom = impl.createDocument(None, "div", None)
+    top = dom.documentElement
+
+    # Set 0 -> None to disable slice limits (show all)
+    max_indexes = config.max_display_indexes or None
+    max_events = config.max_display_events or None
+    max_features = config.max_display_features or None
+
+    # Other configs
+    convert_datetime = evset.schema.is_unix_timestamp
+    visible_features = evset.schema.features[:max_features]
+    index_schemas = evset.schema.indexes
+    all_index_keys = evset.get_index_keys(sort=True)
+    total_indexes = len(all_index_keys)
+    total_features = len(evset.schema.features)
+    hiding_feats = max_features is not None and total_features > max_features
+
+    # Create one table and header per index value
+    for index_key in all_index_keys[:max_indexes]:
+        # Index header text (n events x n features)
+        index_text = ", ".join(
+            [
+                f"{idx.name}={repr_value(val, idx.dtype)}"
+                for val, idx in zip(index_key, index_schemas)
+            ]
+        )
+        index_data = evset.data[index_key]
+        index_n_events = len(index_data.timestamps)
+        title = dom.createElement("h3")
```

Comment:
Note (for later). In case the style (e.g. h3) is already set (which, it likely the case when rending our documentation), I would not use any h1-h3 style. Instead, we could create our own minimalistic css style.